pkg/kubelet/pleg/generic.go

```diff
@@ -122,34 +165,66 @@ func (g *GenericPLEG) relist() {
 		return
 	}
 
+	// Build a new map.
+	podMap := buildPodMap(pods)
```

Comment:
There is not much to gain if we need to clear a map before reusing it (or is it?).

I chose to create a new map each time so that I can more easily compare the new state with the existing state. If you think this would cause too much memory GC action, I can perhaps change it to using only one map and modify the entries in it. WDYT?